### PR TITLE
chore: bump major on Pillow, optional dependency

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY

Bump Pillow optional dependency (used for taking web screenshots) to address several security vulnerabilities: https://www.cvedetails.com/product/27460/Python-Pillow.html?vendor_id=10210

### TESTING INSTRUCTIONS
Make sure we are still able to generate thumbnails and reports with charts and dashboards screenshots

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def get_git_sha() -> str:
             "snowflake-sqlalchemy==1.2.4"
         ],  # PINNED! 1.2.5 introduced breaking changes requiring sqlalchemy>=1.4.0
         "teradata": ["sqlalchemy-teradata==0.9.0.dev0"],
-        "thumbnails": ["Pillow>=7.0.0, <8.0.0"],
+        "thumbnails": ["Pillow>=8.2.0, <9.0.0"],
```
Comment (villebro) [inline setup.py:160]: It appears we're already pinning to >=8.3.1 in the development deps - perhaps we could use the same version here and potentially consider bumping to 8.3.2 which addresses CVE-2021-23437?
https://github.com/apache/superset/blob/0f3b6304549c52dee6b6614446ba3ae09548f526/requirements/development.in#L21

**Review by villebro — APPROVED** [review]
LGTM with a non-blocking consideration

File: setup.py (hunk 1 above)
Comment (dpgaspar) [reply setup.py]: good point, made the change

**Review by villebro — APPROVED** [review]
LGTM

**Pull request merged**